omc3/correction/handler.py

```diff
@@ -109,6 +119,60 @@ def correct(accel_inst: Accelerator, opt: DotDict) -> None:
     LOG.info("Finished Iterative Global Correction.")
 
 
+
+def _update_response(
+    accel_inst: Accelerator, 
+    corrected_model: pd.DataFrame,
+    corrected_elements: pd.DataFrame,
+    delta: pd.DataFrame, 
+    optics_params: Sequence[str], 
+    variable_categories: Sequence[str], 
+    update_response: bool | str,
+    ):
+    """ Create an updated response matrix.
+    
+    If we are to compute the response including the DPP, then we have to do so from MAD-X, 
+    as we do not have the analytical formulae.
+    Otherwise we go through the way of computing the response the user requested.
+
+    In case of dpp being modified in the correction, we have to add the new dpp value
+    to the accelerator instance, to run the `twiss` in madx at this dpp.
+    All other parameters are taken care of in the model/elements.
+    The dpp value has to be reset afterwards, as in the calculation of the correction,
+    the delta is added via the CHANGEPARAMTERS file.
+
+    Alternatively: We could create a copy of the accelerator instance,
+    but that would require a proper copy implementation of the class, I think (jdilly 2024)
+    """
+    # update model, not nice as "private" elements, but will be reset later
+    original_model = accel_inst.model 
+    original_elements = accel_inst.elements
+
+    accel_inst._model = corrected_model
+    accel_inst._elements = corrected_elements
+    
+    update_dpp = ORBIT_DPP in delta.index
+
+    if update_dpp:
+        LOG.info("Updating response via MAD-X, due to delta dpp requested.")
+        dpp = delta.loc[ORBIT_DPP, DELTA]
+        accel_inst.dpp += dpp # Add the delta dpp to be used in the twisses of the FR creation
+        resp_dict = response_madx.create_fullresponse(accel_inst, variable_categories)
+        accel_inst.dpp -= dpp # Reset the dpp
+    else:
+        if update_response == "madx":
+            LOG.info("Updating response via MAD-X.")
+            resp_dict = response_madx.create_fullresponse(accel_inst, variable_categories)
+        else:
+            LOG.info("Updating response via analytical formulae.")
+            resp_dict = response_twiss.create_response(accel_inst, variable_categories, optics_params)
+    
+    # reset model (not really necessary as only used here, but nicer)
+    accel_inst._model = original_model
+    accel_inst._elements = original_elements
+    return resp_dict
```

Comment:
Since we do not return anything to do with the "accelerator", wouldn't it be better to just make a copy of the `accel_inst` (i.e. via `copy.deepcopy`) and use that copy as argument in the function calls here? This way there is no worry about resetting the right attributes in the `accel_inst`.

Pingging @JoschD to get his opinion as well.

Reply:
Otherwise the function here is quite nice, happy with it.